### ISSUE TITLE
Flexibility for drivers to set their rates and areas served

### ISSUE THREAD

**Issue by Aetos4617** [p2prider/p2prider#12]

**Comment by Aetos4617** [comment]
All drivers may not like to be paid the meter fare so i think there could be an option where we give the drivers to give their own fares along a range that could be bounded with a max amt according to the distance to be travelled

**Comment by amoghamblihalli** [comment]
thats a good idea. i think rather than to rely only on distance to calculate, we can also use that day's fuel prices, average traffic along those routes, etc.  to ensure a balanced maximum.

**Comment by ashwingattani** [comment]
Or we can ask the driver to set the base meter rate and calculate based on that? The driver can alter the base rate at any point in time, giving them the ability to earn depending on how the day moves.

**Comment by ashwingattani** [comment]
> 

Implementing this in the MVP would be a tedious and complicated job. Also, the fuel prices rarely change on regular basis, it happens only during certain global/political shifts, calculating based on distance, similarly how the auto meters work would be simple and make more sense.

**Comment by amoghamblihalli** [comment]
fair enough. although for the driver to set the base might set a bad precedent

**Comment by ashwingattani** [comment]
@67void We can provide a range from which drivers can select the base price. I don't think drivers can use it to make customers pay more than the market standards.

**Comment by amoghamblihalli** [comment]
yea thats what i meant. but the way that range is calculated takes into those multiple factors

**Comment by ashwingattani** [comment]
Agreed, we'll need to brainstorm how that can be done.

**Comment by p2prider** [comment]
To summarize, there are a few factors that we need to work with here and device an algorithm which takes these into account in a fair way to both drivers and riders:

1. Fuel price
2. Traffic condition on the route
3. Favorability of the route
4. Driver's and rider's expected rates per km
5. Minimum and maximum rates applicable in the area

This algorithm should also be simple for users to understand. We don't want a situation where users are trying to game the algorithm treating it like a black box. It should not smell of trickery, it should paint a picture of complete transparency and trust. This has been one of the biggest issues with ola/uber for drivers, and fixing this will give our project a lot of credibility among them and make onboarding easier.

The simplest two options to tackle here are the last two: I think it makes sense to let drivers and riders both set their expected rates per km inside a given range. It's a cheap and effective filter. It gives enough leeway for free-market economics to work while also setting a ceiling to prevent exorbitant pricing. I'm in favor of doing this for the MVP itself. (But the community should vote!)

Second is a little bit difficult simply because of implementation challenges. We can't use OSM for this, and google maps API is expensive.

Third one is somewhat doable. We can provide an option for drivers to select polygons or streets on the map and set specific rates on them. This might be a bit technically challenging so we should probably hold it from the MVP and ask for feedback from drivers and tell them that we'd be able to do this in the future.

The first one - fuel price - is definitely doable.

Regardless, after more discussion on the individual properties of these factors, we should devise a simple algorithm to combine them.
